packages/backend/src/open_payments/receiver/service.ts

```diff
@@ -179,3 +188,49 @@ async function getLocalIncomingPayment({
 
   return incomingPayment.toOpenPaymentsType({ ilpStreamConnection: connection })
 }
+
+async function getIncomingPaymentGrant(
+  deps: ServiceDependencies,
+  paymentPointerUrl: string
+): Promise<Grant | undefined> {
+  const paymentPointer = await deps.openPaymentsClient.paymentPointer.get({
+    url: paymentPointerUrl
+  })
+  if (!paymentPointer) {
+    return undefined
+  }
+  const grantOptions = {
+    authServer: paymentPointer.authServer,
+    accessType: AccessType.IncomingPayment,
+    accessActions: [AccessAction.ReadAll]
+  }
+
+  const existingGrant = await deps.grantService.get(grantOptions)
+  if (existingGrant) {
+    return existingGrant
+  }
+
+  const grant = await deps.openPaymentsClient.grant.request({
```

Comment:
Is it worth (get or creating and) locking the auth server's db row here to avoid simultaneously sending redundant grant requests?

Reply:
Is this a case for when we request the receiver on the outgoing payment lifecycle hook?

Reply:
I was thinking if (multiple instances of) the backend were handling two quotes with the same receiver at the same time, should it avoid performing multiple grant requests?

Reply:
Pretty edge-casey IMO, not sure if worth adding just yet